src/Microsoft.AspNet.Mvc.Xml/Formatters/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -106,6 +106,13 @@ protected virtual XmlReader CreateXmlReader([NotNull] Stream readStream)
         /// <returns>The <see cref="XmlObjectSerializer"/> used during deserialization.</returns>
         protected virtual XmlObjectSerializer CreateDataContractSerializer(Type type)
         {
+            // check if the type to which we are deserliazing is of type SerializableError
+            // if yes, then use the SerializableErrorWrapper instead.
```

Comment:
comment should say why we are doing this,  not what we are doing (what can be read, why is not obvious).